tests/test_aadt_mapping.py

```diff
@@ -40,6 +44,36 @@ def test_automapper(external_loop, arch_fc):
             assert ir_bv()(**ir_inputs) == arch_bv()(**arch_inputs)[0]
 
 
+@pytest.mark.parametrize('external_loop', [True, False])
+def test_reg(external_loop):
+    IR = gen_SmallIR(32)
+    family = regsim.family
+    arch_bv = regsim.RegPE_mappable_fc(family.PyFamily())
+    arch_mapper = ArchMapper(regsim.RegPE_mappable_fc)
+    expect_found = frozenset(('Add', 'Nor', 'Not'))
+    expect_not_found = IR.instructions.keys() - expect_found
+
+    for ir_name, ir_fc in IR.instructions.items():
+        ir_mapper = arch_mapper.process_ir_instruction(ir_fc)
+        rewrite_rule = ir_mapper.solve('z3', external_loop=external_loop)
+        if rewrite_rule is None:
+            assert ir_name in expect_not_found
+            continue
+        assert ir_name in expect_found
+
+        #verify the mapping works
+        counter_example = rewrite_rule.verify()
+        assert counter_example is None
+        ir_bv = ir_fc(family.PyFamily())
+        ir_paths, arch_paths = rewrite_rule.get_input_paths()
+        # Need register allocations for this to work:
+        #for _ in range(num_test_vectors):
```

Comment:
It would be nice if we got something working to verify it works in Py land, good for now